### PR TITLE
feat(chat): integrate with publication API (Issue #1269)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

Integrate with publication API 

Issues:

- #1269

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/473012)

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/474692)
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/474698)
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review

>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Chat/ChatExternalControls.tsx [hunk 1]
```diff
@@ -7,22 +7,43 @@ import { ConversationInfo } from '@/src/types/chat';
...
 
 export default function ChatExternalControls({ conversations }: Props) {
   const { t } = useTranslation(Translation.Chat);
+
   const dispatch = useAppDispatch();
 
+  const resourceToReview = useAppSelector((state) =>
+    PublicationSelectors.selectResourceToReviewByReviewUrl(
+      state,
+      conversations[0].id,
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatExternalControls.tsx]:
```suggestion
      conversations[0]?.id,
```

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+} from '@/src/store/publication/publication.reducers';
+
+import CaretIconComponent from '../../Common/CaretIconComponent';
+import CollapsibleSection from '../../Common/CollapsibleSection';
+import {
+  ConversationPublicationResources,
+  PromptPublicationResources,
+} from './PublicationResources';
+
+import { some } from 'lodash-es';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]:
```suggestion
import some from 'lodash-es/some';
```

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+              }),
+            );
+          }
+
+          dispatch(
+            ConversationsActions.selectConversations({
+              conversationIds: [],
+            }),
+          );
+        }}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: at least it should be moved in `useCallback`

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        className={classNames(
+          'group relative flex h-[30px] items-center rounded border-l-2 hover:bg-accent-primary-alpha',
+          selectedPublication?.url === publication.url &&
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx:90]: Are you sure that we need to use property `url` instead of `id` here and in other places?
I believe we can everywhere use `id` for consistency with other entities (maybe interface `Entity` with `status`)

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      : PromptPublicationResources;
+
+  return (
+    <div className="flex flex-col gap-1">
+      <div
+        onClick={() => {
+          setIsOpen(true);
+          if (!isOpen) {
+            dispatch(
+              PublicationActions.uploadPublication({ url: publication.url }),
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: I don't think that we need to upload publication each time.
Please, look at property `status` in interface `Entity`

Comment (Alexander-Kezik) [reply apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: I only upload a publication if the publication itself has been closed. I don't go after it more than once

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx (hunk 4 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx:90]: publications do not have id, they have url. we could convert the data, like url to id, but I'm not sure that's a good idea

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx (hunk 5 above)
Comment (IlyaBondar) [reply apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: @Alexander-Kezik ,
Ohh, why I can't close opened publication? It's bug
Every section on our UI can be expanded/collapsed and publication also should.
And uploading should happens only in first expanding

Comment (IlyaBondar) [reply apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: will be fixed here
https://github.com/epam/ai-dial-chat/issues/1430

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+          <CaretIconComponent isOpen={isOpen} />
+          <IconClipboard className="text-secondary" width={18} height={18} />
+          <div
+            className={classNames(
+              'relative max-h-5 flex-1 truncate break-all text-left',
+              some(publication.resources, (r) =>
+                some(selectedConversationIds, (id) =>
+                  id.startsWith(r.reviewUrl),
+                ),
+              ) && 'text-accent-primary',
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: it would be nice to move in constant. Ideally with `useMemo`

File: apps/chat/src/utils/app/data/publication-service.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  public static getRules(
+    path: string,
+  ): Observable<{ rules: Record<string, PublicationRule[]> }> {
+    return ApiUtils.request('api/publication/rulesList', {
+      method: 'POST',
+      body: JSON.stringify({ url: path ? `public/${path}/` : `public/` }),
+    });
+  }
+
+  public static getSelectedConversationsId(): Observable<string | null> {
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/data/publication-service.ts]:
```suggestion
  public static getSelectedPublicationId(): Observable<string | null> {
```

File: apps/chat/src/utils/app/data/publication-service.ts [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+      body: JSON.stringify({ url: ApiUtils.encodeApiUrl(url) }),
+    });
+  }
+
+  public static getRules(
+    path: string,
+  ): Observable<{ rules: Record<string, PublicationRule[]> }> {
+    return ApiUtils.request('api/publication/rulesList', {
+      method: 'POST',
+      body: JSON.stringify({ url: path ? `public/${path}/` : `public/` }),
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/data/publication-service.ts]: `ApiUtils.encodeApiUrl` here ?

File: apps/chat/src/utils/app/data/publication-service.ts [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    parentPath: string,
+    entityType: ApiKeys,
+    options?: Partial<{ recursive: boolean }>,
+  ): Observable<PublishedItem> {
+    const query = new URLSearchParams({
+      ...(options?.recursive && { recursive: String(options.recursive) }),
+    });
+    const resultQuery = query.toString();
+    return ApiUtils.request(
+      `api/publication/${entityType}/public/${parentPath}?${resultQuery}`,
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/data/publication-service.ts]: `parentPath` is encoded here?

File: apps/chat/src/pages/index.tsx [hunk 10]
```diff
@@ -326,6 +326,9 @@ export const getServerSideProps: GetServerSideProps = async ({
           ? Feature.HideNewConversation
           : [],
       ),
+    publicationFilters: (process.env.PUBLICATION_FILTERS || 'title,role').split(
```
Comment (IlyaBondar) [inline apps/chat/src/pages/index.tsx:329]: Why do we set `title,role` by default here?

File: apps/chat/src/types/publication.ts [hunk 11]
```diff
@@ -0,0 +1,103 @@
+import { BackendDataNodeType, BackendResourceType } from './common';
+
+export enum PublicationFunctions {
+  EQUAL = 'EQUAL',
+  CONTAIN = 'CONTAIN',
+  REGEX = 'REGEX',
```
Comment (IlyaBondar) [inline apps/chat/src/types/publication.ts]: According to Figma filters should be `Contains`, `Regex`, `Equal` and `Not contains`.
Why are they in uppercase here?
And how are you going to support `UserGroup` here?
image

File: apps/chat/src/store/publication/publication.reducers.ts [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+      state,
+      _action: PayloadAction<{
+        targetFolder: string;
+        resources: { targetUrl: string }[];
+      }>,
+    ) => state,
+    deletePublicationFail: (state) => state,
+    uploadPublishedWithMeItems: (
+      state,
+      _action: PayloadAction<{ featureType: ApiKeys }>,
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.reducers.ts]: Use `FeatureType` instead
```suggestion
      _action: PayloadAction<{ featureType: FeatureType }>,
```
and `getApiKeyByFeatureType` in `getPublishedWithMeItems`

File: apps/chat/src/utils/app/publications.ts [hunk 13]
```diff
@@ -0,0 +1,31 @@
+import { ApiKeys } from '@/src/types/common';
+import { SharingType } from '@/src/types/share';
+
+import { constructPath } from './file';
+
+export const isItemPublic = (id: string) => id.split('/')[1] === 'public';
+
+export const createTargetUrl = (
+  apiKey: ApiKeys,
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/publications.ts]: Use `featureType: FeatureType` instead
```suggestion
  featureType: FeatureType,
```
and `getApiKeyByFeatureType` on service level.
Avoid to use `ApiKeys` in our logic

File: apps/chat/src/components/Chat/Publish/PublishWizard.tsx [hunk 14]
```diff
@@ -423,11 +500,17 @@ export default function PublishWizard({
...
           setIsChangeFolderModalOpened(false);
         }}
         type={type}
         depth={depth}
-        rootFolderId={getFileRootId()}
+        rootFolderId={
+          type === SharingType.Conversation ||
+          type === SharingType.ConversationFolder
+            ? constructPath(ApiKeys.Conversations, 'public')
+            : constructPath(ApiKeys.Prompts, 'public')
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/PublishWizard.tsx]: Avoid to use `ApiKeys` in our logic and use existing util-methods istead of own construction of paths
```suggestion
            ? getRootId({ featureTupe: FeatureType.Chat , bucket: 'public')
            : getRootId({ featureTupe: FeatureType.Prompt , bucket: 'public')
```

File: apps/chat/src/store/publication/publication.epics.ts [hunk 15]
```diff
@@ -0,0 +1,755 @@
...
+    filter(PublicationActions.publish.match),
+    switchMap(({ payload }) => {
+      const encodedTargetFolder = ApiUtils.encodeApiUrl(payload.targetFolder);
+      const targetFolderSuffix = payload.targetFolder ? '/' : '';
+
+      return PublicationService.publish({
+        targetFolder: constructPath(
+          `public${payload.targetFolder ? '' : '/'}`,
+          `${encodedTargetFolder}${targetFolderSuffix}`,
+        ),
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.epics.ts]:
```suggestion
        targetFolder: `${constructPath(
          'public',
          encodedTargetFolder,
        )}${targetFolderSuffix}`,
```

File: apps/chat/src/store/publication/publication.epics.ts [hunk 16]
```diff
@@ -0,0 +1,755 @@
...
+
+const deletePublicationEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(PublicationActions.deletePublication.match),
+    switchMap(({ payload }) => {
+      const encodedTargetFolder = ApiUtils.encodeApiUrl(payload.targetFolder);
+      const targetFolderSuffix = payload.targetFolder ? '/' : '';
+
+      return PublicationService.deletePublication({
+        targetFolder: `${encodedTargetFolder}${targetFolderSuffix}`,
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.epics.ts:309]: Why don't we have `public/` prefix here as we have in `publishEpic`?

File: apps/chat/src/store/publication/publication.epics.ts [hunk 17]
```diff
@@ -0,0 +1,755 @@
...
+    mergeMap(({ payload }) =>
+      PublicationService.getPublishedWithMeItems('', payload.featureType).pipe(
+        mergeMap((publications) => {
+          const actions: Observable<AnyAction>[] = [];
+          const selectedIds =
+            ConversationsSelectors.selectSelectedConversationsIds(state$.value);
+
+          if (
+            selectedIds.some((id) =>
+              id.startsWith(constructPath(ApiKeys.Conversations, 'public/')),
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.epics.ts]: use `getRootId` with `bucket: 'public'` instead

File: apps/chat/src/store/publication/publication.epics.ts [hunk 18]
```diff
@@ -0,0 +1,755 @@
...
+            )
+          ) {
+            const pathsToUpload = selectedIds.filter((id) =>
+              id.startsWith(constructPath(ApiKeys.Conversations, 'public/')),
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.epics.ts]: use `getRootId` with `bucket: 'public'` instead

File: apps/chat/src/store/publication/publication.epics.ts [hunk 19]
```diff
@@ -0,0 +1,755 @@
...
+                ),
+              ),
+            );
+          }
+
+          if (!publications.items) {
+            return EMPTY;
+          }
+
+          if (payload.featureType === ApiKeys.Conversations) {
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.epics.ts]:
```suggestion
          if (payload.featureType === FeatureType.Chat) {
```

File: apps/chat/src/store/publication/publication.epics.ts [hunk 20]
```diff
@@ -0,0 +1,755 @@
...
+                        model: parsedApiKey.model,
+                        name: parsedApiKey.name,
+                        publishedWithMe: true,
+                      };
+                    }),
+                  }),
+                ),
+              );
+            }
+          } else if (payload.featureType === ApiKeys.Prompts) {
```
Comment (IlyaBondar) [inline apps/chat/src/store/publication/publication.epics.ts]:
```suggestion
          } else if (payload.featureType === FeatureType.Prompt) {
```

File: apps/chat/src/pages/api/publication/rulesList.ts [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+
+import fetch from 'node-fetch';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const session = await getServerSession(req, res, authOptions);
+  const isSessionValid = validateServerSession(session, req, res);
+  const token = await getToken({ req });
+  if (!isSessionValid) {
+    return;
+  }
```
Comment (IlyaBondar) [inline apps/chat/src/pages/api/publication/rulesList.ts]:
```suggestion
  if (!isSessionValid) {
    return;
  }
  const token = await getToken({ req });
```

File: apps/chat/src/pages/api/publication/resourceListing.ts [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+
+import fetch from 'node-fetch';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const session = await getServerSession(req, res, authOptions);
+  const isSessionValid = validateServerSession(session, req, res);
+  const token = await getToken({ req });
+  if (!isSessionValid) {
+    return;
+  }
```
Comment (IlyaBondar) [inline apps/chat/src/pages/api/publication/resourceListing.ts]: as I can see we shouldn't read token before session checking here in other handlers
```suggestion
  if (!isSessionValid) {
    return;
  }
  const token = await getToken({ req });
```

File: apps/chat/src/components/Search/SearchFiltersView.tsx [hunk 23]
```diff
@@ -59,16 +59,6 @@ export default function SearchFiltersView({
           dataQa: 'shared-by-me-filter',
           filterValue: SearchFilters.SharedByMe,
         },
-        {
```
Comment (IlyaBondar) [inline apps/chat/src/components/Search/SearchFiltersView.tsx:62]: Why did you delete this filter?

File: apps/chat/src/components/Common/FolderContextMenu.tsx [hunk 24]
```diff
@@ -140,27 +136,16 @@ export const FolderContextMenu = ({
         onClick: onPublishUpdate,
         disabled: disableAll,
       },
-      {
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/FolderContextMenu.tsx:143]: Why did you remove it?

File: apps/chat/src/components/Chatbar/Conversation.tsx [hunk 25]
```diff
@@ -679,6 +678,7 @@ export const ConversationComponent = ({ item: conversation, level }: Props) => {
       {isPublishing && (
         <PublishModal
           entity={conversation}
+          entities={[conversation]}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chatbar/Conversation.tsx:686]: Why do we need it if we have `entity={conversation}`?

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/478967)

File: apps/chat/src/utils/app/data/publication-service.ts (hunk 7 above)
Comment (Alexander-Kezik) [reply apps/chat/src/utils/app/data/publication-service.ts]: Thanks! It was unused code. I removed it

File: apps/chat/src/types/publication.ts (hunk 11 above)
Comment (Alexander-Kezik) [reply apps/chat/src/types/publication.ts]: API doesn't support `Not contains` filter. Figma just represents how it should look like. For user groups, we can just add a new filter and pass it to the rules.

File: apps/chat/src/pages/index.tsx (hunk 10 above)
Comment (Alexander-Kezik) [reply apps/chat/src/pages/index.tsx:329]: We haven't really discussed this. Now I just added these values to be able to see it on a staging environment before we change the configuration to be able to test it with Auth0.

File: apps/chat/src/store/publication/publication.reducers.ts (hunk 12 above)
Comment (Alexander-Kezik) [reply apps/chat/src/store/publication/publication.reducers.ts:79]: This is where I was rather misleading. I changed the name of the variable. Why pass featureType to get apikey if we can just pass apiKey?

File: apps/chat/src/store/publication/publication.epics.ts (hunk 16 above)
Comment (Alexander-Kezik) [reply apps/chat/src/store/publication/publication.epics.ts:309]: we already have `public/` prefix inside id, since it was published

File: apps/chat/src/components/Search/SearchFiltersView.tsx (hunk 23 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Search/SearchFiltersView.tsx:62]: we haven't support it right now

File: apps/chat/src/pages/api/publication/resourceListing.ts (hunk 22 above)
Comment (Alexander-Kezik) [reply apps/chat/src/pages/api/publication/resourceListing.ts]: I fixed it everywhere

File: apps/chat/src/pages/api/publication/rulesList.ts (hunk 21 above)
Comment (Alexander-Kezik) [reply apps/chat/src/pages/api/publication/rulesList.ts]: I fixed it everywhere

File: apps/chat/src/components/Common/FolderContextMenu.tsx (hunk 24 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Common/FolderContextMenu.tsx:143]: I deleted it during development. I'll get it back when I make a folders unpublishing ticket

File: apps/chat/src/components/Chatbar/Conversation.tsx (hunk 25 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Chatbar/Conversation.tsx:686]: this was done to throw child entities if we call this window for folders. And with entities array we can have single interface to map file and resources for folders/conversations/prompts.

File: apps/chat/src/utils/app/data/publication-service.ts [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+    parentPath: string,
+    featureType: FeatureType,
+    options?: Partial<{ recursive: boolean }>,
+  ): Observable<PublishedItem> {
+    const query = new URLSearchParams({
+      ...(options?.recursive && { recursive: String(options.recursive) }),
+    });
+    const resultQuery = query.toString();
+    return ApiUtils.request(
+      `api/publication/${EnumMapper.getApiKeyByFeatureType(featureType)}/public/${ApiUtils.encodeApiUrl(parentPath)}?${resultQuery}`,
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/data/publication-service.ts]: Use `constructPath` instead because I suppose `parentPath` can be empty

File: apps/chat/src/components/Chatbar/Conversation.tsx (hunk 25 above)
Comment (IlyaBondar) [reply apps/chat/src/components/Chatbar/Conversation.tsx:686]: In this case let's remove the property `entity={conversation}` and put main resource as first in `entities`

File: apps/chat/src/components/Search/SearchFiltersView.tsx (hunk 23 above)
Comment (IlyaBondar) [reply apps/chat/src/components/Search/SearchFiltersView.tsx:62]: Then just comment it with TODO-comment to not lose

File: apps/chat/src/components/Common/FolderContextMenu.tsx (hunk 24 above)
Comment (IlyaBondar) [reply apps/chat/src/components/Common/FolderContextMenu.tsx:143]: Then just comment it with TODO-comment to not lose

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/types/publication.ts (hunk 11 above)
Comment (IlyaBondar) [reply apps/chat/src/types/publication.ts]: Why are they in uppercase here?

File: apps/chat/src/components/Chatbar/Conversation.tsx (hunk 25 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Chatbar/Conversation.tsx:686]: This will make it very difficult to understand the code, since entities can contain both a folder and a chat or prompt

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Common/ItemContextMenu.tsx [hunk 27]
```diff
@@ -268,26 +266,11 @@ export default function ItemContextMenu({
         onClick: onPublish,
         disabled: disableAll,
       },
-      {
-        name: t('Update'),
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ItemContextMenu.tsx:272]: Don't we need `Update` option? It's like `Publish` but for `published by me` items

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1314.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/ItemContextMenu.tsx (hunk 27 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Common/ItemContextMenu.tsx:272]: reverted with TODO comment

**Review by IlyaBondar — APPROVED** [review]
